### DISCUSSION TITLE
r chunk option: tidy = styler option planned?

### DISCUSSION THREAD

**Discussion by Ifitrains** [quarto-dev/quarto-cli#3118] [Feature Requests]
You "may use the styler package ([Müller and Walthert 2022](https://bookdown.org/yihui/rmarkdown-cookbook/opts-tidy.html#ref-R-styler)) to reformat your R code if you set the chunk option tidy = 'styler'" (https://bookdown.org/yihui/rmarkdown-cookbook/opts-tidy.html). 

So far in Quarto, I can only set the chunk option `tidy` to true or false. Is this planned?

**Comment by mcanouil** [comment]
FYI @cderv

**Comment by cderv** [comment]
> So far in Quarto, I can only set the chunk option tidy to true or false. Is this planned?

Are you referring to linting red marker in source code inside RStudio IDE ? 

The feature is a **knitr** feature and it is already working. 

````markdown
---
title: "test"
format: html
---

```{r}
#| tidy: styler
a <- 1#one variable
abc <- 2#another variable
```

````

This will have the comments in R code have spaces added by **styler**. However, our YAML linter is indeed not aware of that. This is why there is a red marker inside the document. Though I can render without error. 

We'll fix it.

**Reply by cderv** [reply]
PR is at #6565

**Comment by cderv** [comment]
Styler is now listed in YAML autocompletion for `tidy` option

**Comment by Ifitrains** [comment]
Great, thank you very much for your work!
